components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3111,8 +3117,20 @@ private ClaimConfig getClaimConfiguration(int applicationId, Connection connecti
             while (loadClaimConfigsResultSet.next()) {
                 claimConfig.setRoleClaimURI(loadClaimConfigsResultSet.getString(1));
                 claimConfig.setLocalClaimDialect("1".equals(loadClaimConfigsResultSet.getString(2)));
-                claimConfig.setAlwaysSendMappedLocalSubjectId("1".equals(loadClaimConfigsResultSet
-                        .getString(3)));
+
+                switch (loadClaimConfigsResultSet.getString(3)) {
+                    case "1":
+                        claimConfig.setAlwaysSendMappedLocalSubjectId(true);
+                        claimConfig.setMappedLocalSubjectMandatory(false);
+                        break;
+                    case "2":
+                        claimConfig.setAlwaysSendMappedLocalSubjectId(true);
+                        claimConfig.setMappedLocalSubjectMandatory(true);
+                        break;
+                    default:
```

Comment:
although this may work here we should avoid this kind of branching. Reason is, when setting we take the '2' scenario as the default, and here we are taking '0' as default. If in future some new state is introduces, this can lead to bugs/confusions. Ideally this 'default' block should match the 'else' block in the set path

Reply:
addressed with [0790f6a](https://github.com/wso2/carbon-identity-framework/pull/5110/commits/0790f6a70e5ae0c4b11ec0f2d7246f0baf99b882). Updated the set path to match